pkg/cmd/roachtest/tests/restore.go

```diff
@@ -478,6 +482,35 @@ func registerRestore(r registry.Registry) {
 	}
 }
 
+// skipIfNetworkFlake skips the test if the job error contains a network flake.
+// This function should only get called after a job terminates unexpectedly.
+func skipIfNetworkFlake(ctx context.Context, t test.Test, c cluster.Cluster, jobId int) error {
+	db, err := c.ConnE(ctx, t.L(), c.Node(1)[0])
```

Comment:
Should we `defer db.Close()` here?

Reply:
the go compiler does not appreciate your suggestion:
```
compilepkg: nogo: errors found by nogo during build-time code analysis:
pkg/cmd/roachtest/tests/restore.go:489:2: should check returned error before deferring db.Close() (SA5001)
```